src/app/pages/account/user-details/user-details.component.ts

```diff
@@ -1,33 +1,106 @@
-import { Component, Input, OnInit } from '@angular/core';
+import {Component, OnInit} from '@angular/core';
 import {
   faIdCard,
   faCalendar,
   faEnvelope,
 } from '@fortawesome/free-regular-svg-icons';
-import { faPhone } from '@fortawesome/free-solid-svg-icons';
-import { User } from 'src/app/models/User';
-import { faPen } from '@fortawesome/free-solid-svg-icons';
+import {faPhone} from '@fortawesome/free-solid-svg-icons';
+import {User} from 'src/app/models/User';
+import {faPen} from '@fortawesome/free-solid-svg-icons';
+import {AbstractControl, FormControl, FormGroup, ValidationErrors, Validators} from "@angular/forms";
+import {UserService} from "../../../services/user-service.service";
+import {UserProfile} from "../../../models/user-profile";
 
 @Component({
   selector: 'app-user-details',
   templateUrl: './user-details.component.html',
   styleUrls: ['../account.component.css'],
 })
 export class UserDetailsComponent implements OnInit {
-  @Input() user: User = {
+  user: User/* = {
```

Comment:
This was removed and information passing is now done through a subject because the information from the parent component is not being passed to the child component before the child component html was parsed (which shouldn't be the case but it was happening). Because of this, the formgroup in the child component that is being used for updating user profile information was receiving default values (the part I commented out) unless explicitly changed in the form. So to avoid this I changed all user detailed to be passed through a subject to guarantee everything is happening in the right order.